typescript/ws-protocol-examples/src/examples/mcap-record.ts

```diff
@@ -0,0 +1,221 @@
+    this.handle = handle;
+  }
+
+  async write(buffer: Uint8Array): Promise<void> {
+    const written = await this.handle.write(buffer);
+    this.totalBytesWritten += written.bytesWritten;
+  }
+
+  position(): bigint {
+    return BigInt(this.totalBytesWritten);
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/promise-function-async
+function delay(durationMs: number) {
+  return new Promise((resolve) => setTimeout(resolve, durationMs));
+}
+
+async function waitForServer(
+  address: string,
+  signal: AbortSignal,
+): Promise<FoxgloveClient | undefined> {
+  log("connecting to %s", address);
+  while (!signal.aborted) {
+    const maybeClient = await new Promise<FoxgloveClient | undefined>((resolve) => {
+      const ws = new WebSocket(address, [FoxgloveClient.SUPPORTED_SUBPROTOCOL]);
+      const client = new FoxgloveClient({ ws });
+      const onClose = (event: CloseEvent) => {
+        log(
+          "connection failed, code=%s reason=%s wasClean=%s",
+          event.code,
+          event.reason,
+          event.wasClean,
+        );
+        resolve(undefined);
+      };
+      client.on("close", onClose);
+      client.on("open", () => {
+        client.off("close", onClose);
+        signal.addEventListener("abort", () => {
+          client.close();
+        });
+        resolve(client);
+      });
+    });
+    if (maybeClient) {
+      return maybeClient;
+    }
+    log("trying again in 5 seconds...");
+    await delay(5000);
+  }
+  return undefined;
+}
+
+async function main(address: string, options: { output: string }): Promise<void> {
+  await Zstd.isLoaded;
+  await fs.mkdir(path.dirname(options.output), { recursive: true });
+  const fileHandle = await fs.open(options.output, "w");
+  const fileHandleWritable = new FileHandleWritable(fileHandle);
+  const textEncoder = new TextEncoder();
+
+  const writer = new McapWriter({
+    writable: fileHandleWritable,
+    compressChunk: (data) => ({
+      compression: "zstd",
+      compressedData: Zstd.compress(data, 19),
+    }),
+  });
+
+  await writer.start({
+    profile: "",
+    library: "mcap-record",
+  });
+
+  const controller = new AbortController();
+  process.on("SIGINT", () => {
+    log("shutting down...");
+    controller.abort();
+  });
+
+  try {
+    const client = await waitForServer(address, controller.signal);
+    if (!client) {
+      return;
+    }
+    await new Promise<void>((resolve) => {
+      const wsChannelsByMcapChannel = new Map<McapChannelId, WsChannelId>();
+      const subscriptionsById = new Map<
+        SubscriptionId,
+        { messageCount: number; mcapChannelId: McapChannelId }
+      >();
+
+      client.on("serverInfo", (event) => {
+        log(event);
+      });
+      client.on("status", (event) => {
+        log(event);
+      });
+      client.on("error", (err) => {
+        log("server error: %o", err);
+      });
+
+      client.on("advertise", (newChannels) => {
+        void Promise.all(
```

Comment:
is the `void` necessary here? What purpose does it serve?

Reply:
mostly to keep the linter happy, since we are mixing async and non-async functions here